owasp-top10-2017-apps/a3/snake-pro/app/main.go

```diff
@@ -71,10 +72,26 @@ func main() {
 	echoInstance.Renderer = &TemplateRegistry{
 		templates: templates,
 	}
+
+	echoInstance.GET("/safe", func(c echo.Context) error {
+		req := c.Request()
+		format := `
+			<code>
+				Protocol: %s<br>
+				Host: %s<br>
+				Remote Address: %s<br>
+				Method: %s<br>
+				Path: %s<br>
+			</code>
+		`
+		return c.HTML(http.StatusOK, fmt.Sprintf(format, req.Proto, req.Host, req.RemoteAddr, req.Method, req.URL.Path))
+	})
+
 	echoInstance.GET("/healthcheck", api.HealthCheck)
 	echoInstance.POST("/register", api.Register)
 	echoInstance.POST("/login", api.Login)
 	echoInstance.GET("/login", api.PageLogin)
+	echoInstance.Logger.Fatal(echoInstance.StartTLS(":10033", "cert.pem", "key.pem"))
```

Comment:
This line should be at after all routes declarations, and just be careful, you're starting the API once here and twice in `line 106`

Reply:
Thanks, Daniel. I fixed it at my last PR.